perceval/backends/core/zulip.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2020 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Authors:
+#     Venu Vardhan Reddy Tekula <venuvardhanreddytekula8@gmail.com>
+#
+
+
+import logging
+
+from grimoirelab_toolkit.uris import urijoin
+
+from ...backend import (Backend,
+                        BackendCommand,
+                        BackendCommandArgumentParser)
+from ...client import HttpClient
+
+logger = logging.getLogger(__name__)
+
+CATEGORY_MESSAGE = "message"
+
+
+class Zulip(Backend):
+    """Zulip Backend."""
+
+    version = '0.1.0'
+
+    CATEGORIES = [CATEGORY_MESSAGE]
+
+    def __init__(self, url, stream, api_token, email, origin, tag=None, archive=None, ssl_verify=True):
+        super().__init__(origin, tag, archive, ssl_verify)
+        origin = url
+
+        super.__init__(origin, tag=tag, archive=archive, ssl_verify=True)
+        self.stream = stream
+        self.api_token = api_token
+        self.email = email
+
+        self.client = None
+
+    def fetch(self, category=CATEGORY_MESSAGE):
```

Comment:
The fetch signature misses the offset. The default offset should be declared as constant, which value should be the first offset possible (e.g., DEFAULT_OFFSET = 1)

```
-    def fetch(self, category=CATEGORY_MESSAGE):
+    def fetch(self, category=CATEGORY_MESSAGE, offset=DEFAULT_OFFSET):
         """Fetch the message from the stream."""
-        kwargs = {}
+
+        kwargs = {"offset": offset}
         items = super().fetch(category, **kwargs)
```